token-metadata/js/package.json

```diff
@@ -0,0 +1,83 @@
+{
+    "name": "@solana/spl-token-metadata",
+    "description": "SPL Token Metadata Interface JS API",
+    "version": "0.0.1",
+    "author": "Solana Labs Maintainers <maintainers@solanalabs.com>",
+    "repository": "https://github.com/solana-labs/solana-program-library",
+    "license": "Apache-2.0",
+    "type": "module",
+    "sideEffects": false,
+    "engines": {
+        "node": ">=16"
+    },
+    "files": [
+        "lib",
+        "src",
+        "LICENSE",
+        "README.md"
+    ],
+    "publishConfig": {
+        "access": "public"
+    },
+    "main": "./lib/cjs/index.js",
+    "module": "./lib/esm/index.js",
+    "types": "./lib/types/index.d.ts",
+    "exports": {
+        "types": "./lib/types/index.d.ts",
+        "require": "./lib/cjs/index.js",
+        "import": "./lib/esm/index.js"
+    },
+    "scripts": {
+        "build": "tsc --build --verbose tsconfig.all.json",
+        "clean": "shx rm -rf lib **/*.tsbuildinfo || true",
+        "deploy": "npm run deploy:docs",
+        "deploy:docs": "npm run docs && gh-pages --dest token-metadata/js --dist docs --dotfiles",
+        "docs": "shx rm -rf docs && typedoc && shx cp .nojekyll docs/",
+        "fmt": "prettier --write '{*,**/*}.{ts,tsx,js,jsx,json}'",
+        "lint": "prettier --check '{*,**/*}.{ts,tsx,js,jsx,json}' && eslint --max-warnings 0 .",
+        "lint:fix": "npm run fmt && eslint --fix .",
+        "nuke": "shx rm -rf node_modules package-lock.json || true",
+        "postbuild": "shx echo '{ \"type\": \"commonjs\" }' > lib/cjs/package.json",
+        "reinstall": "npm run nuke && npm install",
+        "release": "npm run clean && npm run build",
+        "test": "mocha test",
+        "watch": "tsc --build --verbose --watch tsconfig.all.json"
+    },
+    "peerDependencies": {
+        "@solana/web3.js": "^1.47.4"
+    },
+    "dependencies": {
+        "@solana/codecs-core": "2.0.0-experimental.398c396",
+        "@solana/codecs-data-structures": "2.0.0-experimental.398c396",
+        "@solana/codecs-numbers": "2.0.0-experimental.398c396",
+        "@solana/codecs-strings": "2.0.0-experimental.398c396",
+        "@solana/options": "2.0.0-experimental.398c396",
+        "@solana/spl-type-length-value": "0.1.0"
+    },
+    "devDependencies": {
+        "@solana/web3.js": "^1.47.4",
```

Comment:
Is this supposed to be _both_ a `devDependency` and `peerDependency`?

Reply:
Yes - I believe so. 

It needs to be a `devDependency ` so it exists when you running tests inside library, but when the library it is imported, it will be a `peerDependency` and use the version that already exists. 

Its the same setup as here as well:
https://github.com/solana-labs/solana-program-library/blob/master/token/js/package.json